extensions/functions_rounding.yaml

```diff
@@ -27,3 +27,68 @@ scalar_functions:
           - value: fp64
             name: "x"
         return: fp64
+  -
+    name: "round"
+    description: >
+      Rounding the value `x` to s decimal places.
+    impls:
+      - args:
+          - value: fp32
+            name: "x"
+            description: >
+              Numerical expression to be rounded.
+          - value: i32
+            name: "s"
+            description: >
+              Number of decimal places to be rounded to.
+
+              When `s` is a positive number, the rounding
+              is performed to a `s` number of decimal places.
+
+              When `s` is a negative number, the rounding is
+              performed to the left side of the decimal point
+              as specified by `s`.
+          - value: i32
+            name: "function"
+            description: >
+              This is an optional argument `function`, a numeric
+              value determines whether to perform round or truncate.
+              When the `function` value is set to 0, the function
+              performs rounding to `s` decimal places, otherwise it
+              performs truncate. The default value is 0.
```

Comment:
Why isn't this an enum?

Reply:
@westonpace I think roughly I get your point. But could you please give me an example here, I don't think my understanding is very precise.